app/resources/essentials.py

```diff
@@ -101,12 +101,15 @@ def delete_one_service(service: Service):
 def register_service(device_uuid: str, service_uuid: str, name: str, user: str) -> float:
     r_data: dict = {"device_uuid": device_uuid, "service_uuid": service_uuid}
 
-    given_per: Tuple[float, float, float, float, float] = game_content.dict2tuple(
-        m.contact_microservice(
-            "device", ["hardware", "register"], {**r_data, **config["services"][name]["needs"], "user": user}
-        )
+    r_ms_data: dict = m.contact_microservice(
+        "device", ["hardware", "register"], {**r_data, **config["services"][name]["needs"], "user": user}
     )
 
+    if "error" in r_ms_data:
+        return r_ms_data
```

Comment:
The `register_service` function should return the new speed of the service. If the service could not be registered maybe we should return `-1` or something or just throw an exception and handle it in the endpoints. Also add a new unit test for this case.